openid-4-verifiable-presentations-1_0.md

```diff
@@ -549,11 +555,273 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
 
 If the Verifier responds with any HTTP error response, the Wallet MUST terminate the process.
 
+# Verifiable Presentation Query Language {#vp_query}
+
+The Verifiable Presentation Query Language (VP Query) is a JSON-encoded query
+language that allows the Verifier to request the Wallet to present Verifiable
+Credentials that match the query. The Verifier MAY encode constraints on the
+combinations of credentials and claims that may be returned. The Wallet evaluates the
+query against the Verifiable Credentials it holds and returns the Verifiable
+Presentations that match the query.
+
+A valid VP Query is defined as a JSON-encoded object with the following
+top-level properties:
+
+`expect_credentials`:
+: REQUIRED. A non-empty array of Credential Queries as defined in (#credential_query)
+that specify the requested Verifiable Credentials.
+
+`valid_credential_sets`:
+: OPTIONAL. A non-empty array containing arrays of identifiers for elements in
+`expect_credentials` that defines which sets of Credentials may be returned.
+The identifier MAY be postfixed by `?`, indicating that
+delivery of the the respective credential is optional. For details, see
+(#vp_query_lang_processing_rules).
+
+Note: While this specification does not define additional top-level properties,
+future extensions MAY define additional properties that modify the behavior of
+the Wallet when processing the query.
+
+## Credential Query {#credential_query}
+
+A Credential Query is an object representing a request for one specific
+credential. 
+
+It contains the following properties:
+
+`id`:
+: REQUIRED. A string identifying the credential in the response and, if provided,
+the constraints in `valid_credential_sets`.
```

Comment:
Does this need the constraint to not have duplicate ids?

Reply:
Yes, I'll add that.